test_utils.go

```
@@ -0,0 +1,261 @@
+		return nil, nil, nil, nil, err
+	}
+	bobCommitTx, err := lnwallet.CreateCommitTx(fundingTxIn, bobKeyPub,
+		aliceKeyPub, bobRevokeKey, csvTimeoutBob, channelBal, channelBal, bobDustLimit)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	alicePath, err := ioutil.TempDir("", "alicedb")
+	dbAlice, err := channeldb.Open(alicePath)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	bobPath, err := ioutil.TempDir("", "bobdb")
+	dbBob, err := channeldb.Open(bobPath)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	var obsfucator [lnwallet.StateHintSize]byte
+	copy(obsfucator[:], aliceFirstRevoke[:])
+
+	estimator := &lnwallet.StaticFeeEstimator{FeeRate: 50}
+	feePerKw := btcutil.Amount(estimator.EstimateFeePerWeight(1) * 1000)
+	aliceChannelState := &channeldb.OpenChannel{
+		IdentityPub:            aliceKeyPub,
+		ChanID:                 prevOut,
+		ChanType:               channeldb.SingleFunder,
+		FeePerKw:               feePerKw,
+		IsInitiator:            true,
+		StateHintObsfucator:    obsfucator,
+		OurCommitKey:           aliceKeyPub,
+		TheirCommitKey:         bobKeyPub,
+		Capacity:               channelCapacity,
+		OurBalance:             channelBal,
+		TheirBalance:           channelBal,
+		OurCommitTx:            aliceCommitTx,
+		OurCommitSig:           bytes.Repeat([]byte{1}, 71),
+		FundingOutpoint:        prevOut,
+		OurMultiSigKey:         aliceKeyPub,
+		TheirMultiSigKey:       bobKeyPub,
+		FundingWitnessScript:   witnessScript,
+		LocalCsvDelay:          csvTimeoutAlice,
+		RemoteCsvDelay:         csvTimeoutBob,
+		TheirCurrentRevocation: bobRevokeKey,
+		RevocationProducer:     alicePreimageProducer,
+		RevocationStore:        shachain.NewRevocationStore(),
+		TheirDustLimit:         bobDustLimit,
+		OurDustLimit:           aliceDustLimit,
+		Db:                     dbAlice,
+	}
+
+	addr := &net.TCPAddr{
+		IP:   net.ParseIP("127.0.0.1"),
+		Port: 18555,
+	}
+
+	aliceChannelState.SyncPending(addr, 0)
```

The error is ignored here.